superset/connectors/sqla/models.py

```diff
@@ -1106,20 +1106,22 @@ def get_sqla_row_level_filters(
         filter_groups: Dict[Union[int, str], List[TextClause]] = defaultdict(list)
         try:
             for filter_ in security_manager.get_rls_filters(self):
-                clause = self.text(
-                    f"({template_processor.process_template(filter_.clause)})"
-                )
-                if filter_.group_key:
-                    filter_groups[filter_.group_key].append(clause)
-                else:
-                    all_filters.append(clause)
+                if filter_.clause:
+                    clause = self.text(
+                        f"({template_processor.process_template(filter_.clause)})"
+                    )
+                    if filter_.group_key:
+                        filter_groups[filter_.group_key].append(clause)
+                    else:
+                        all_filters.append(clause)
```

Comment:
Looking at `SupersetSecurityManager.get_rls_filters(self)`, it states it should be returning `List[SqlaQuery]`. However, it appears to be returning `List[RowLevelSecurityFilter]`. So I believe that should also be fixed.